### DISCUSSION TITLE
"Start Activity" synchronously?

### DISCUSSION THREAD

**Discussion by don-jaime** [gopro/labs#901] [Feature Requests]
On Labs, I've been mainly enjoying long time lapse photo sets (e.g., `!47NQmPN!S!1R`). A caveat is that, in low light, this can result in an exposure that's longer than the `!1R` delay-and-repeat at the end of the script, and I'd like to easily avoid a longer static delay. I think a workaround is already available by delaying longer at certain time windows, but I wonder if it's feasible to avoid heuristics like that.

1. Does it even make sense for me to _want_ to accomplish this? I.e., is it even worthwhile to try to get night photo mode (`mPN`) time lapse frames _at_ night (to try to get some stars and pre-dawn clouds in my all-day time lapses)?
2. If so, then I'd like to have a synchronous "start activity" action (even if only for photos, where the activity is expected to naturally end within a relatively short time). That way, I'd know the script would not continue until the photo was finished, _and_ I wouldn't have to set a longer delay that would unnecessarily burn battery faster when it's not needed.

Again, I realize I could probably work around this conditionally based on the clock (or maybe even doing something slick with metering to detect the light level and calculating a delay based on that?), but I'm hoping for a really foolproof way to get the same result by just telling my script to wait until each frame (photo) is done. I've learned a good bit while trying to understand this over the last couple of years, and if I missed an existing feature, I apologize that I must not have thought of the right search terms.

**Comment by dnewman-gpsw** [comment]
If you are using HERO11 or 12 this should do the trick.
![1000007383-01](https://github.com/gopro/labs/assets/10856084/cb544ad8-af7c-465e-9606-224267fae014)
![Nigh Photo every minute -5_10_2024, 13_33_35](https://github.com/gopro/labs/assets/10856084/0d72dd4a-457d-4a88-bd55-eb1ee18b4cad)

**Reply by dnewman-gpsw** [reply]
This will trigger whenever the time seconds are zero, so it will not be an issue for longer exposures.
